python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedAdministrationCheck.java

```diff
@@ -43,146 +44,14 @@
-          map.addIssue(CIDRIP, MESSAGE);
-          map.addIssue(CIDRIPV6, MESSAGE);
-        }
-      });
-  }
-
-  private static boolean isDictionaryWithAttributeBadProtocolEmptyIpAddressAdminPort(DictionaryAsMap map) {
-    return map.hasKeyValuePair(IPPROTOCOL, isString(SENSITIVE_PROTOCOL))
-      && (map.hasKeyValuePair(CIDRIP, isString(EMPTY_IPV4)) || map.hasKeyValuePair(CIDRIPV6, isString(EMPTY_IPV6)))
-      && map.hasSensitivePortRange("fromPort", "toPort");
-  }
-
-  private static boolean isDictionaryWithAttributeInvalidProtocolEmptyIpAddress(DictionaryAsMap map) {
-    return map.hasKeyValuePair(IPPROTOCOL, isString(ANY_PROTOCOL))
-      && (map.hasKeyValuePair(CIDRIP, isString(EMPTY_IPV4)) || map.hasKeyValuePair(CIDRIPV6, isString(EMPTY_IPV6)));
-  }
-
-  // Class to handle Dictionary elements as a Map of ResolvedKeyValuePair, useful in case we need to refer to several elements and not only looking for a specific one.
-  // The keys are all resolved String in CdkUtils.ResolvedKeyValuePair.key
-  static class DictionaryAsMap {
-    Map<String, CdkUtils.ResolvedKeyValuePair> map = new HashMap<>();
-
-    public static DictionaryAsMap build(SubscriptionContext ctx, DictionaryLiteral dictionary) {
-      DictionaryAsMap dict = new DictionaryAsMap();
-
-      List<CdkUtils.ResolvedKeyValuePair> pairs = dictionary.elements().stream()
-        .map(e -> CdkUtils.getKeyValuePair(ctx, e))
-        .filter(Optional::isPresent).map(Optional::get).collect(Collectors.toList());
-
-      for (CdkUtils.ResolvedKeyValuePair pair : pairs) {
-        pair.key.getExpression(isStringLiteral()).map(StringLiteral.class::cast)
-          .ifPresent(key -> dict.map.put(key.trimmedQuotesValue(), pair));
-      }
-      return dict;
-    }
-
-    public boolean hasKeyValuePair(String key, Predicate<Expression> valuePredicate) {
-      return map.containsKey(key) && map.get(key).value.hasExpression(valuePredicate);
-    }
-
-    public Optional<Expression> get(String key, Predicate<Expression> valuePredicate) {
-      if (!map.containsKey(key)) {
-        return Optional.empty();
-      }
-      return map.get(key).value.getExpression(valuePredicate);
-    }
-
-    public void addIssue(String key, String message) {
-      if (map.containsKey(key)) {
-        map.get(key).value.addIssue(message);
-      }
-    }
-
-    public Optional<Long> getArgumentAsLong(String name) {
-      return get(name, isNumericLiteral())
-        .map(NumericLiteral.class::cast)
-        .map(NumericLiteral::valueAsLong);
-    }
-
-    boolean hasSensitivePortRange(String minName, String maxName) {
-      Optional<Long> min = getArgumentAsLong(minName);
-      Optional<Long> max = getArgumentAsLong(maxName);
-
-      if (min.isEmpty() || max.isEmpty()) {
-        return false;
-      }
-
-      return isInInterval(min.get(), max.get(), ADMIN_PORTS);
-    }
```

Comment:
It would have been nice to split this refactoring in a separate commit, to ease review